packages/jazz-tools/src/tools/implementation/zodSchema/zodReExport.ts

```diff
@@ -41,3 +51,57 @@ export {
   type output as infer,
   z,
 } from "zod/v4";
+
+type NonCoZodType = core.$ZodType & { collaborative?: false };
+
+export function record<
+  Key extends core.$ZodRecordKey,
+  Value extends NonCoZodType,
+>(
+  keyType: Key,
+  valueType: Value,
+  params?: string | core.$ZodRecordParams,
+): ZodRecord<Key, Value> {
+  return zodRecord(keyType, valueType, params);
+}
+
+export function optional<T extends NonCoZodType>(schema: T): ZodOptional<T> {
+  return zodOptional(schema);
+}
+
+export function discriminatedUnion<
+  T extends readonly [
+    NonCoZodType & core.$ZodTypeDiscriminable,
+    ...(NonCoZodType & core.$ZodTypeDiscriminable)[],
+  ],
+>(discriminator: string, schemas: T): ZodDiscriminatedUnion<T> {
+  return zodDiscriminatedUnion(discriminator, schemas as any);
+}
```

Comment:
Better to keep optional and discriminatedUnion out for now.

Not support interop with Co schemas would be a breaking change because it works currently.